main.go

```diff
@@ -63,6 +63,14 @@ func buildLibrary() fyne.CanvasObject {
 		if c, ok := current.(*overlayContainer); ok {
 			if selected != nil {
 				c.c.Objects = append(c.c.Objects, wrapContent(selected.create()))
+				var localPackageList = make(map[string]bool)
+				for _, v := range selected.pacakges {
+					localPackageList[v] = true
+				}
+				for k := range packageList {
+					localPackageList[k] = true
+				}
+				packageList = localPackageList
```

Comment:
I would think you could do this without a global variable - by walking the tree at export time perhaps?

Reply:
Yes, that's true. But this distributes the load and seemed simpler as it did not require me to learn walking the tree. Do you recommend that walking the tree would be much simpler than whatever is being done here?

Reply:
Maintaining state is much more complex than simply calculating it once (as you noted with the need to check import counts etc. None of that is needed if you just walk the structure (which is quite simple, you just need to check if something is a container and if so check it's child elements.

Reply:
It's not just the containers that we'll have to check, right? We need to check it with multiple elements. For example., if there is a button, we need to check if has an icon or not, and that requires two different packages - widget and theme. Or let's say, there is a list - in which case, we will have to import container, fyne, layout, theme and widget in our default setup. It seemed better to do it this way instead of the tree walking.

Reply:
Yes, for each widget you would need to ask it what imports are required (probably another callback function).
However I don't think this can be avoided because `Button` just like `TabContainer` etc can be used with or without `theme` package so we can't make assumptions.

Reply:
Instead, wouldn't it be easy if we just do it the way we're doing right now?

Reply:
Well, globals are something to avoid in general - if we have multiple windows it will break and it is much harder to maintain.
Add to this that many widgets don't have a single definition of imports (for example Button with or without icon is different).

Maintaining a static list that then has to have unused removed afterwards seems less efficient than getting the correct import set when exporting with a tree walk.